Programmig with objects/index.html

```diff
@@ -0,0 +1,92 @@
+                </p>
+        </div>
+        <div class="concept">
+            <!-- Making classes -->
+            <div class="concept-title" id="lesson-3-3">
+                <b>Making classes</b>
+            </div>
+            <p>A class defines a data type, which contains variables, properties and methods. For example, we can make a class by bringing the function <span class="code">__init__</span> and listing what we want our class to capture. In this case we create a class <em>Movie</em> and we want it to remember:
+            <ul>
+            <li>title</li>
+            <li>storyline</li>
+            <li>poster_image</li>
+            <li>trailer_youtube</li>
+            </ul>So we create this class which functions as a blue print and we cant make different instances of this class like we made one of toy story and avatar.
+            </p>
+            <p>When we made the class Movie we made some <b>instance variables</b> so when we called the class <em>Movie</em> we need to give the variables: title, story line, poster image, and trailer.
+            </p>
+        </div>
+        <div class="concept">
+            <!-- Inheritance and Abstraction -->
+            <div class="concept-title" id="lesson-3-4">
+                <b>Inheritance, Abstraction and how it relates</b>
+            </div>
+            <p><b>Inheritance</b> We use inheritance for resolving attribute references: if a requested attribute is not found in the DerivedClass, the search proceeds to look in the BaseClass. This rule is applied recursively if the base class itself is derived from some other class. For example we can create a class called Video's which will be the BaseClass then we can have two DerivedClasses the class movies, that we have already written, and then create a new class called Tv-Shows. Then the DerivedClasses will get some attributes from the BaseClass if not already defined within that instance.
+            </p>
+            <p><b>Abstraction</b> is the process of taking away or removing chracteristics from something in order to reduce it to a set of essential characteristics. Throught the process of abstraction, a programmer hides all but the relevant data about an object in order to reduce complexity and increse efficiency.
+            </p>
+            <p>In OOP we use classes to help us write code faster and more efficiently. We can relate this to the same code we have used already by building this website using HTML and CSS. For example we have classes in our HTML code called "concept" then in CSS we define what characteristics we want our class to have. This make things much easier because instead of defining what we want each paragraph or section to look like we can make a class then just put all of the sections we want to have the same characteristics.
+            </p>
```

Comment:
I'm not sure which notes are yours, if any. Make reference to the where you get your information and **write your own notes**.